dpctl/tests/test_usm_ndarray_dlpack.py

```diff
@@ -234,3 +247,53 @@ def test_dlpack_from_subdevice():
     ar = dpt.arange(n, dtype=dpt.int32, sycl_queue=q)
     ar2 = dpt.from_dlpack(ar)
     assert ar2.sycl_device == sdevs[0]
+
+
+def test_legacy_dlpack_capsule():
+    try:
+        x = dpt.arange(100, dtype="i4")
+    except dpctl.SyclDeviceCreationError:
+        pytest.skip("No default device available")
+
+    cap = x.__dlpack__(max_version=(0, 8))
+    y = _dlp.from_dlpack_capsule(cap)
+    del cap
+    assert x._pointer == y._pointer
+
+    x2 = dpt.reshape(x, (10, 10)).mT
+    cap = x2.__dlpack__(max_version=(0, 8))
+    y = _dlp.from_dlpack_capsule(cap)
+    del cap
+    assert x2._pointer == y._pointer
+
+    x3 = x[::-2]
+    cap = x3.__dlpack__(max_version=(0, 8))
+    y = _dlp.from_dlpack_capsule(cap)
+    assert x3._pointer == y._pointer
+    del x3, y, x
+    del cap
+
+
+def test_versioned_dlpack_capsule():
+    try:
+        x = dpt.arange(100, dtype="i4")
+    except dpctl.SyclDeviceCreationError:
+        pytest.skip("No default device available")
+
+    cap = x.__dlpack__(max_version=(1, 0))
```

Comment:
May be a good idea to replace `max_version=(1, 0)` calls with the result of `get_build_dlpack_version`, so they don't have to be changed in the future.

Reply:
Actually, thinking about it more, never mind—it's fine like this. As long as the build version is higher than (1, 0) it will work, and this specifically tests the versioned capsule, which may be a subset of future functionality.